File: med/cashRegister/cashRegister.rb
File.open(ARGV[0]).each do |line|
	line = line.split(';').map(&:to_f)
	price = line[0].round(2)
	paid = line[1].round(2)
	answer = ''	

	if price > paid
		answer += 'ERROR,'
		change = 0.00
	end
	if price == paid
		answer += 'ZERO,'
		change = 0.00
	end

	if paid > price
		change = paid - price
	end

	if change >= 99.999
		answer += 'ONE HUNDRED,'
		change -= 100.00
	end
	
	if change >= 49.999
		answer += 'FIFTY,'
		change -= 50.00
	end

	if change >= 19.999
		answer += 'TWENTY,'
		change -= 20.00
	end

	if change >= 19.999
		answer += 'TWENTY,'
		change -= 20.00
	end

	if change >= 9.999
		answer += 'TEN,'
		change -= 10.00
	end

	if change >= 4.999
		answer += 'FIVE,'
		change -= 5.00
	end

	if change >= 1.999
		answer += 'TWO,'
		change -= 2.00
	end

	if change >= 1.999
		answer += 'TWO,'
		change -= 2.0
	end

	if change >= 0.999
		answer += 'ONE,'
		change -= 1.00
	end

	if change >= 0.499
		answer += 'HALF DOLLAR,'
		change -= 0.50
	end

	if change >= 0.249
		answer += 'QUARTER,'
		change -= 0.25
	end

	if change >= 0.099
		answer += 'DIME,'
		change -= 0.10
	end

	if change >= 0.099
		answer += 'DIME,'
		change -= 0.10
	end

	if change >= 0.049
		answer += 'NICKEL,'
		change -= 0.05
	end

	if change >= 0.009
		answer += 'PENNY,'
		change -= 0.01
	end

	if change >= 0.009
		answer += 'PENNY,'
		change -= 0.01
	end


	if change >= 0.009
		answer += 'PENNY,'
		change -= 0.01
	end


	if change >= 0.009
		answer += 'PENNY,'
		change -= 0.01
	end
	
	answer = answer.chomp(',')
	
	puts answer
end

File: codeeval/med/cashRegister/cashRegister.cs
using System;
using System.IO;

class program
{
    static void Main(string[] args)
    {
        //using (StreamReader sr = new StreamReader("test_file.txt"))
        using (StreamReader sr = File.OpenText(args[0]))
        {
            string line;
            program a = new program();

            while ((line = sr.ReadLine()) != null)
            {
                a.findAnswer(line);
            }
        }
        Console.Read();
    }

    public void findAnswer(string line)
    {
        double pp, ch, change;
        string answer = "";

        pp = Convert.ToDouble(line.Substring(0, line.IndexOf(';')));
        ch = Convert.ToDouble(line.Substring(line.IndexOf(';')+ 1));

        change = Math.Round(ch - pp, 2);

        if (pp > ch)
            answer = "ERROR";
        else if (pp == ch)
            answer = "ZERO";


        if (change >= 100.0)
        {
            answer += "ONE HUNDRED,";
            change = Math.Round(change - 100.0, 2);
        }

        if (change >= 50.0)
        {
            answer += "FIFTY,";
            change = Math.Round(change - 50.0, 2);
        }

        if (change >= 20.0)
        {
            answer += "TWENTY,";
            change = Math.Round(change - 20.0, 2);
        }

        if (change >= 20.0)
        {
            answer += "TWENTY,";
            change = Math.Round(change - 20.0, 2);
        }

        if (change >= 10.0)
        {
            answer +=  "TEN,";
            change = Math.Round(change - 10.0, 2);
        }

        if (change >= 5.0)
        {
            answer += "FIVE,";
            change = Math.Round(change - 5.0, 2);
        }

        if (change >= 2.0)
        {
            answer += "TWO,";
            change = Math.Round(change - 2.0, 2);
        }

        if (change >= 2.0)
        {
            answer += "TWO,";
            change = Math.Round(change - 2.0, 2);
        }

        if (change >= 1.0)
        {
            answer += "ONE,";
            change = Math.Round(change - 1.0, 2);
        }

        if (change >= 0.50)
        {
            answer += "HALF DOLLAR,";
            change = Math.Round(change - 0.50, 2);
        }

        if (change >= 0.25)
        {
            answer += "QUARTER,";
            change = Math.Round(change - 0.25, 2);
        }

        if (change >= 0.10)
        {
            answer += "DIME,";
            change = Math.Round(change - 0.10, 2);
        }

        if (change >= 0.10)
        {
            answer += "DIME,";
            change = Math.Round(change - 0.10, 2);
        }

        if (change >= 0.05)
        {
            answer += "NICKEL,";
            change = Math.Round(change - 0.05, 2);
        }

        if (change >= 0.01)
        {
            answer += "PENNY,";
            change = Math.Round(change - 0.01, 2);
        }

        if (change >= 0.01)
        {
            answer += "PENNY,";
            change = Math.Round(change - 0.01, 2);
        }

        if (change >= 0.01)
        {
            answer += "PENNY,";
            change = Math.Round(change - 0.01, 2);
        }

        if (change >= 0.01)
        {
            answer += "PENNY,";
            change = Math.Round(change - 0.01, 2);
        }
        if (answer.Substring(answer.Length - 1) == ",")
            answer = answer.Substring(0, answer.Length - 1);

        Console.WriteLine(answer);

    }
}



